ecore/go/src/pkg/ecore/elist_impl.go

```diff
@@ -0,0 +1,275 @@
+}
+
+// InsertAll element of an array at a given position
+func (arr *ArrayEList) InsertAll(index int, list EList) bool {
+	if index < 0 || index > arr.Size() {
+		panic("Index out of bounds")
+	}
+	for it := list.Iterate(); (*it).Next(); {
+		arr.Insert(index, (*it).Value())
+		index++
+	}
+	return true
+}
+
+// Move an element to the given index
+func (arr *ArrayEList) Move(index int, elem interface{}) {
+	me := arr.IndexOf(elem)
+	if me == -1 {
+		panic("Index out of bounds")
+	}
+	arr.Swap(me, index)
+}
+
+// Swap move an element from oldIndex to newIndex
+func (arr *ArrayEList) Swap(oldIndex, newIndex int) interface{} {
+	if oldIndex < 0 || oldIndex >= arr.Size() ||
+		newIndex < 0 || newIndex > arr.Size() {
+		panic("Index out of bounds")
+	}
+	val := arr.Get(oldIndex)
+	arr.RemoveAt(oldIndex)
+	if newIndex > oldIndex {
+		newIndex--
+	}
+	arr.Insert(newIndex, val)
+	return val
+}
+
+// Get an element of the array
+func (arr *ArrayEList) Get(index int) interface{} {
+	if index < 0 || index >= arr.Size() {
+		panic("Index out of bounds")
+	}
+	return arr.data[index]
+}
+
+// Set an element of the array
+func (arr *ArrayEList) Set(index int, elem interface{}) {
+	if index < 0 || index >= arr.Size() {
+		panic("Index out of bounds")
+	}
+	arr.data[index] = elem
+}
+
+// RemoveAt remove an element at a given position
+func (arr *ArrayEList) RemoveAt(index int) bool {
+	if index < 0 || index >= arr.Size() {
+		panic("Index out of bounds")
+	}
+	arr.data = append(arr.data[:index], arr.data[index+1:]...)
+	return true
+}
+
+// Remove an element in an array
+func (arr *ArrayEList) Remove(elem interface{}) bool {
+	index := arr.IndexOf(elem)
+	if index == -1 {
+		panic("Index out of bounds")
+	}
+	return arr.RemoveAt(index)
+}
+
+// Size count the number of element in the array
+func (arr *ArrayEList) Size() int {
+	return len(arr.data)
+}
+
+// Clear remove all elements of the array
+func (arr *ArrayEList) Clear() {
+	arr.data = make([]interface{}, 0)
+}
+
+// Empty return true if the array contains 0 element
+func (arr *ArrayEList) Empty() bool {
+	return arr.Size() == 0
+}
+
+// Contains return if an array contains or not an element
+func (arr *ArrayEList) Contains(elem interface{}) bool {
+	return arr.IndexOf(elem) != -1
+}
+
+// IndexOf return the index on an element in an array, else return -1
+func (arr *ArrayEList) IndexOf(elem interface{}) int {
+	index := 0
+	for it := arr.Iterate(); (*it).Next(); {
+		if (*it).Value() == elem {
+			return index
+		}
+		index++
+	}
+	return -1
+}
+
+// Iterate through the array
+func (arr *ArrayEList) Iterate() *EIterator {
+	var it EIterator
+	myIt := arrayEListIterator{data: *arr, curr: -1}
+	it = &myIt
+	return &it
+}
+
+// Value return the current value of the iterator
+func (it *arrayEListIterator) Value() interface{} {
+	return it.data.Get(it.curr)
+}
+
+// Next make the iterator go further in the array
+func (it *arrayEListIterator) Next() bool {
+	it.curr++
+	if it.curr == it.data.Size() {
```

Comment:
equivalent a return it.curr == it.data.Size()
Ca evite un test

Reply:
Fix with 0c24124b7228dc0abc4999e664f2bad3540968fd